File: benchmarks/oopsla25/subjects/431.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f431(const char *s) {
    assert(strlen(s) == 1);
    char c = s[0];
    if (c != 'a') {
        if (c != 'c') {
            assert(c != 'b');
        }
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f431(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/431.py
#!/usr/bin/env python3

import sys

def f431(s: str):
  assert len(s) == 1
  c = s[0]
  if c != "a":
    if c != "c":
      assert c != "b"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f431(my_string)
